src/tests/ngraph_helpers/ngraph_functions/src/augru_cell.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vector>
+#include <memory>
+
+#include "ngraph_functions/builders.hpp"
+
+#include "ngraph_ops/augru_cell.hpp"
+#include "ngraph_ops/augru_sequence.hpp"
+
+namespace ngraph {
+namespace builder {
+
+/**
+ * There are 2 options to paramter "in" when "make_sequence" is true.
+ * 0          1               2           3
+ * X   init_hidden_state  attention    seq_length
+ * or,
+  * 0          1               2
+ * X   init_hidden_state   attention
+ * 
+ */
+std::shared_ptr<ov::Node> makeAUGRU(const OutputVector& in,
+                                      const std::vector<ov::Shape>& constants,
+                                      std::size_t hidden_size,
+                                      bool make_sequence,
+                                      ov::op::RecurrentSequenceDirection direction,
+                                      ngraph::helpers::SequenceTestsMode mode) {
+    std::vector<float> empty;
+    auto W = ngraph::builder::makeConstant(in[0].get_element_type(), constants[0], empty, true);
+    W->set_friendly_name("augru_w");
+    auto R = ngraph::builder::makeConstant(in[0].get_element_type(), constants[1], empty, true);
+    R->set_friendly_name("augru_r");
+    auto B = ngraph::builder::makeConstant(in[0].get_element_type(), constants[2], empty, true);
+    B->set_friendly_name("augru_b");
+    if (!make_sequence) {
+        return std::make_shared<ov::op::internal::AUGRUCell>(in[0], in[1], W, R, B, in[2], hidden_size);
+    } else {
+        if (in.size() > 3 && in[3].get_partial_shape().is_dynamic()) {
+            return std::make_shared<ov::op::internal::AUGRUSequence>(in[0], in[1], in[3], W, R, B, in[2], hidden_size);
+        } else {
+            std::shared_ptr<Node> seq_lengths;
+            switch (mode) {
+                case ngraph::helpers::SequenceTestsMode::PURE_SEQ:
+                case ngraph::helpers::SequenceTestsMode::CONVERT_TO_TI_MAX_SEQ_LEN_CONST: {
+                    std::vector<float> lengths(in[0].get_partial_shape()[0].get_min_length(), in[0].get_partial_shape()[1].get_min_length());
```

Comment:
@nshchego @itikhono Do you think it is expected here to use "get_min_length" instead of "get_max_length"? "CONVERT_TO_TI_MAX_SEQ_LEN_CONST" is for maximum, I think.

In grusequence and augrusequence (which is a copy from the previous), the seq_lengths are of 1s. What's more, the results of compute are not matched to the ngraph reference. I think there is bug somewhere. @mbencer

Reply:
Do GRUseq tests covers this case and do they pass?

Reply:
these tests are skipped in GRUseq tests. please cross check comments https://github.com/openvinotoolkit/openvino/pull/12647#discussion_r960383661